src/controllers/auth/index.js

```diff
@@ -16,6 +16,19 @@ exports.signin = async (request, response) => {
   }
 };
 
+exports.signup = async (request, response) => {
+  try {
+    const { vendor, oAuthToken, signupInfo } = request.body;
+
+    const userId = await AuthService.validateOAuthToken(vendor, oAuthToken);
+    const user = await UserService.registerUser({ userId, ...signupInfo });
```

Comment:
함수의 인자로 `User` 객체를 전달해주는건 어떤가요?

```suggestion
    const user = await UserService.registerUser(new User({ userId, ...signupInfo }));
```

객체 값을 받는 것 보다 인스턴스를 받는게 어떤 정보들이 넘어오는지 잘 알 수 있을 것 같습니다.

Reply:
가입을 하고 닉네임은 다음 페이지에서 설정하도록 구현하는 것으로 아는데 그러한 기능은 추후에 추가될 예정인가요?

Reply:
> 객체 값을 받는 것 보다 인스턴스를 받는게 어떤 정보들이 넘어오는지 잘 알 수 있을 것 같습니다.

감사합니다! 반영하겠습니다.

> 가입을 하고 닉네임은 다음 페이지에서 설정하도록 구현하는 것으로 아는데 그러한 기능은 추후에 추가될 예정인가요?

oauth로 가입을 하고 닉네임까지 작성을 한 다음에야 해당 signup API가 호출이 되는 방식입니다! oauth로 로그인하여 얻은 vendor와 oauthToken을 닉네임 작성하는 페이지까지 state로 가져온 다음에, 작성이 끝나고 API가 호출되는 방식입니다.